sessionctx/variable/session.go

```diff
@@ -403,15 +406,14 @@ func (sc *StatementContext) AppendWarning(warn error) {
 
 // HandleTruncate ignores or returns the error based on the StatementContext state.
 func (sc *StatementContext) HandleTruncate(err error) error {
-	if err == nil {
-		return nil
-	}
-	if sc.IgnoreTruncate {
-		return nil
-	}
-	if sc.TruncateAsWarning {
-		sc.AppendWarning(err)
-		return nil
+	if terr, ok := errors.Cause(err).(*terror.Error); ok && terr.Equal(errTruncated) {
```

Comment:
It's better to return the result directly at first. Like the old code on line 406-408. It's the `go` style.